iceoryx_hoofs/test/moduletests/test_posix_shared_memory_object.cpp

```diff
@@ -251,4 +252,79 @@ TEST_F(SharedMemoryObject_Test, OpeningSharedMemoryAndReadMultipleContents)
     EXPECT_THAT(*sutValue1, Eq(4557));
     EXPECT_THAT(*sutValue2, Eq(8912));
 }
+
+#if !defined(_WIN32) && !defined(__APPLE__)
```

Comment:
What do you think about having a `foo_support.hpp` in the platform with defines/constexprs like `#define HAS_BAR true;`? Nothing for this PR but maybe creating an issue for this?

It could be used like this
```cpp
::testing::Test::RecordProperty("TEST_ID", "a9859b5e-555b-4cff-b418-74168a9fd85a");
IOX_TEST_SKIP_IF_NOT_SUPPORTED(HAS_BAR) << "The functionality is not supported by this platform";

auto sut = iox::posix::SharedMemoryObjectBuilder()
...
```

Reply:
I think this is a great idea! I created the issue #1965